lib/api/apiUtils/bucket/bucketDeletion.js

```diff
@@ -111,15 +111,19 @@ export function deleteBucket(bucketMD, bucketName, canonicalID, log, cb) {
 
     return async.waterfall([
         function checkForObjectsStep(next) {
-            return metadata.listObject(bucketName, { maxKeys: 1 }, log,
-                (err, objectsListRes) => {
+            const params = { maxKeys: 1, listingType: 'DelimiterVersions' };
```

Comment:
No. We're using the `DelimiterVersions` extension and `metadata` here to do low-level listing. This will always result in a list of versions and delete markers, regardless of the versioning configuration of the bucket, which is S3's notation.